plugins/offers_offer.c

```diff
@@ -522,12 +530,18 @@ struct command_result *json_offer(struct command *cmd,
 
 	/* BOLT #12:
 	 * - if it includes `offer_paths`:
-	 *...
+	 *   - MAY set `offer_issuer_id`.
 	 * - otherwise:
 	 *   - MUST set `offer_issuer_id` to the node's public key to request the
 	 *     invoice from.
 	 */
-	offer->offer_issuer_id = tal_dup(offer, struct pubkey, &id);
+	/* Although `offer_issuer_id` is not required for payment when using
+	 * blinded paths, it can be forced is requested */
+	if (!we_want_blinded_path(cmd->plugin, false) || *force_issuer_id ) {
+	  offer->offer_issuer_id = tal_dup(offer, struct pubkey, &id);
+	} else {
+	  offer->offer_issuer_id = NULL;
+	}
```

Comment:
just declare it `offer->offer_issuer_id = NULL;` at the biginning of the function